### ISSUE TITLE
unexpected X-Version-Id

### ISSUE THREAD

**Issue by ogmueller** [dmke/mattercheck#2]
I just upgraded to mattermost v8.0 and had mattercheck running against it. There seems to be a problem parsing the version.

```bash
# mattercheck https://mattermost.tld.com
[Jul 18 10:49:31] ERROR mattercheck: could not check instance error=unexpected X-Version-Id, cannot parse "8.0.1..1-rc1.a0ec9ed200f5c39bf4a2324983601795.false" url=https://mattermost.tld.com
```

I am using this mattermost docker image:
MATTERMOST_IMAGE=mattermost-team-edition
MATTERMOST_IMAGE_TAG=release-8.0

**Comment by dmke** [comment]
Yeah, they've botched something when they consolidated their repos before the 8.0.0 release, see https://github.com/mattermost/mattermost/issues/24034.

The latest release of mattercheck specifically handles the exact 8.0.0 release, I was hoping the issue would have been resolved with the [8.0.1 release](https://github.com/mattermost/mattermost/releases/tag/v8.0.1) (I haven't had the chance to update my MM installation, so I can't verify it yet).

N.B. The `release-x.y` Docker tags contain the latest *development* efforts (including beta and RC versions), the *stable* releases are tagged `x.y` and `x.y.z`. Are you intentionally running a potentially unstable version?

**Comment by dmke** [comment]
`mattermost/mattermost-team-edition:8.0` currently responds with `X-Version-Id: 8.0.1.8.0.1.5c4e1e4c4b4074cb17494290436b9285.false`, and mattercheck is able to parse it:

```console
$ mattercheck https://mattermost.example.com
[Jul 27 09:01:18]  INFO mattercheck: instance is up-to-date url=https://mattermost.example.com version=v8.0.1-team
```

I'll monitor the `release-*` tags a bit and see whether (a) they continue to diverge from version format, and (b) I want to add continued support for it in mattercheck.

**Comment by dmke** [comment]
I've reworked the version parsing again, and just released [v0.3.0](https://github.com/dmke/mattercheck/releases/tag/v0.3.0).
